File: DSA/Rahul_Mankani_590014725/Day3_Question3.c
#include <stdio.h>

int SecondMax(int arr[], int n) 
{
    if(n < 2) 
    {
        return -1;
    }
    int max = arr[0];
    int second = -1;
    for(int i = 1; i < n; i++) 
    {
        if(arr[i] > max) 
        {
            second = max;
            max = arr[i];
        }
        else if(arr[i] < max && (second == -1 || arr[i] > second)) 
        {
            second = arr[i];
        }
    }
    return second;
}

int main() 
{
    int n;
    printf("Enter the size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter %d elements: ", n);
    for(int i = 0; i < n; i++) 
    {
        scanf("%d", &arr[i]);
    }
    int result = SecondMax(arr, n);
    if(result == -1) 
    {
        printf("No second largest element exists: -1\n");
    } 
    else 
    {
        printf("Second largest element: %d\n", result);
    }
    return 0;
}

File: DSA/Tanisha_500125283/Day3Question3.java
// Problem Statement: Given an array nums of size n, find the second largest unique element in the array. If it 
// doesn't exist (e.g., all elements are the same), return -1. 
// Task: Find the second largest unique element in the array, or return -1 if it doesn’t exist.
// Examples Input: nums = [5, 5, 5] 
// Output: -1

package DSA.Tanisha_500125283;
import java.util.*;
public class Day3Question3 {
    public static int findingSecondLargest(int[] arr){
        // used hashset because hashset automatically ignores duplicate elements
        Set<Integer> unique = new HashSet<>();
        for(int num:arr){
            unique.add(num);
        }
        
        //used this because if the set has less than 2 unique numbers then we can't find second largest
        if(unique.size()<2){
            return -1;
        }

        // used Integer.MIN_VALUE  to initialize max and secondMax to compare them
        int max= Integer.MIN_VALUE;
        int secondmax=Integer.MIN_VALUE;


        for (int num : unique) {
            if (num > max) {
                secondmax = max;
                max = num;
            } else if (num > secondmax) {
                secondmax = num;
            }
        }

        return secondmax;
    }

    public static void main(String[] args) {
        int[] arr = {5,5,5};
        System.out.println(findingSecondLargest(arr));
    }
}










